Runtime/Scripts/RGBotServerListener.cs

```diff
@@ -559,41 +560,48 @@ private void FixedUpdate()
                                 // set this to avoid expensive lookups next time
-                                rgEntity.ClientId = clientId;
-                                agentMap[clientId].Add(rgEntity);
+                                statefulRGEntity.ClientId = clientId;
+                                agentMap[clientId].Add(statefulRGEntity);
                             }
                         }
                     }
-                }
-                
-                var key = gameObjectState.id.ToString();
-                if (fullGameState.TryGetValue(key, out var combinedGameObjectState))
-                {
-                    // if GameObject has multiple state scripts attached to it,
-                    // then we need to combine their state attributes into one RGStateEntity object.
-                    // iterate over the new fields and add them to the existing entry in the fullGameState.
-                    gameObjectState.ToList().ForEach(x =>
+
+                    // handle merging all the states into 1 overall state for the entity
+                    var key = gameObjectStateEntity.id.ToString();
+                    if (fullGameState.TryGetValue(key, out var combinedGameObjectState))
                     {
-                        // Note that base state info like "position", "rotation", "isPlayer", etc.
-                        // is automatically set to the component's RGEntity values so we don't need to worry about conflicts
-                        if(combinedGameObjectState.ContainsKey(x.Key) && !_duplicatedStateFields.Contains(x.Key))
+                        // if GameObject has multiple state scripts attached to it,
+                        // then we need to combine their state attributes into one RGStateEntity object.
+                        // iterate over the existing fields and add them to the new entry in the fullGameState.
+                        // ... hopefully there is only 1, so we we can have a nice 'typed' RGstateentity in the state, but if multiple.. last one wins :/
+                        // ... you still have all the data, just not easy '.' accessors for everything
+                        combinedGameObjectState.ToList().ForEach(x =>
                         {
-                            RGDebug.LogWarning($"RGEntity with ObjectType {combinedGameObjectState["type"]} has duplicate state attribute {x.Key}");
-                        }
-                        combinedGameObjectState[x.Key] = x.Value;
-                    });
-                    fullGameState[key] = combinedGameObjectState;
-                }
-                else
-                {
-                    fullGameState[key] = gameObjectState;
+                            // Note that core state files like "position", "rotation", "isPlayer", etc are ignored.
+                            // They are part of the core state and can be overridden, but we don't log a warning for it.
```

Comment:
(Future) If it's possible to override these, then we should consider not allowing users to do that. These are keys reserved for very specific concepts - if the user wants to "override" them then they can always define a new key.

Reply:
actually.. in my platformer bot, i found that overriding position was a very logical thing to do.  The player model's position was centered somewhere around his chest/neck, but that was completely useless for sprite tile mapping.  

I re-mapped his position to be relative to the ground at his feet.

Thus why I was quite intentional in putting this code in such that you can override those fields.

I can see an argument for blocking everything but position/rotation from being overridden though.

Reply:
adding code now to block all but those 2 as i rework this section anyway